draft-ietf-gnap-core-protocol.md

```diff
@@ -5890,6 +5892,12 @@ token itself. This approach would make interception of the return from the token
 equivalent to that of a bearer token, since all information required to use the access token
 would be present in the request.
 
+## Key Rotation Policy {#security-key-rotation}
+
+The AS can define its own policy regarding the timeout of the "previous_key", either making it immediately obsolete or allowing for a limited grace period during which both the "previous_key" and the current key can be used for signing requests. Such a grace period can be useful when there are multiple running instances of the client and it's tough, or even impossible, to update the keys on all these instances immediately (just as an example, the client instance could be deployed as a service in kubernetes).
```

Comment:
```suggestion
When keys are rotated, there could be a delay in the propagation of that rotation to various components in the AS's ecosystem. The AS can define its own policy regarding the timeout of the `previous_key`, either making it immediately obsolete or allowing for a limited grace period during which both the `previous_key` and the current key can be used for signing requests. Such a grace period can be useful when there are multiple running copies of the client that are coordinated with each other. For example, the client software could be deployed as a cloud service with multiple orchestrated nodes. Each of these copies is deployed using the same key and therefore all the nodes represent the same client instance to the AS. In such cases, it can be difficult, or even impossible, to update the keys on all these copies in the same instant.
```